java-manta-client-unshaded/src/main/java/com/joyent/manta/config/BaseChainedConfigContext.java

```diff
@@ -446,78 +443,133 @@ public void overwriteWithContext(final ConfigContext context) {
             this.verifyUploads = context.verifyUploads();
         }
+
         if (context.getUploadBufferSize() != null) {
             this.uploadBufferSize = context.getUploadBufferSize();
         }
+
         if (context.getSkipDirectoryDepth() != null) {
             this.skipDirectoryDepth = context.getSkipDirectoryDepth();
         }
+
         if (context.getPruneEmptyParentDepth() != null) {
             this.pruneEmptyParentDepth = context.getPruneEmptyParentDepth();
         }
+
         if (context.downloadContinuations() != null) {
             this.downloadContinuations = context.downloadContinuations();
         }
+
         if (context.getMetricReporterMode() != null) {
             this.metricReporterMode = context.getMetricReporterMode();
         }
+
         if (context.getMetricReporterOutputInterval() != null) {
             this.metricReporterOutputInterval = context.getMetricReporterOutputInterval();
         }
-        if (context.isClientEncryptionEnabled() != null) {
-            this.clientEncryptionEnabled = context.isClientEncryptionEnabled();
-        }
+
         if (context.isContentTypeDetectionEnabled() != null) {
             this.contentTypeDetectionEnabled = context.isContentTypeDetectionEnabled();
         }
+
+        overwriteWithContextEncryptionParams(context);
+
+    }
+
+    /**
+     * Overwrites the configuration timeouts values with values of passed context
+     * if those values are not null and aren't empty.
+     *
+     * @param context context to overwrite configuration with
+     */
+    private void overwriteWithContextTimeouts(final ConfigContext context) {
+        boolean isDefaultContext = context.getClass().equals(DefaultsConfigContext.class);
+
+        if (context.getTimeout() != null) {
+            this.timeout = context.getTimeout();
+        }
+
+        if (context.getTcpSocketTimeout() != null) {
+            this.tcpSocketTimeout = context.getTcpSocketTimeout();
+        }
+
+        if (context.getConnectionRequestTimeout() != null) {
+            this.connectionRequestTimeout = context.getConnectionRequestTimeout();
+        }
+
+        if (context.getExpectContinueTimeout() != null) {
+            this.expectContinueTimeout = context.getExpectContinueTimeout();
+        }
+
+    }
+
+    /**
+     * Overwrites the configuration values with values of passed context
+     * if those values are not null and aren't empty.
+     *
+     * @param context context to overwrite configuration with
+     */
+    private void overwriteWithContextEncryptionParams(final ConfigContext context) {
+        boolean isDefaultContext = context.getClass().equals(DefaultsConfigContext.class);
+
+        if (context.isClientEncryptionEnabled() != null) {
+            this.clientEncryptionEnabled = context.isClientEncryptionEnabled();
+        }
+
         if (context.getEncryptionKeyId() != null) {
             this.encryptionKeyId = context.getEncryptionKeyId();
         }
+
```

Comment:
All these whitespaces have been added to the original file for whitespace consistency.